src/widgets/allergies/allergies.component.tsx

```diff
@@ -1,20 +1,20 @@
 import React from "react";
-import { Switch, Route, useRouteMatch } from "react-router-dom";
+import { Switch, Route, BrowserRouter } from "react-router-dom";
 import AllergyDetailedSummary from "./allergies-detailed-summary.component";
 import AllergyRecord from "./allergy-record.component";
 
 export default function Allergies(props: AllergiesProps) {
-  const match = useRouteMatch();
-  const path = match.url.replace(":subView", "details");
   return (
-    <Switch>
-      <Route exact path={`${path}`}>
-        <AllergyDetailedSummary />
-      </Route>
-      <Route exact path={`${path}/:allergyUuid`}>
-        <AllergyRecord />
-      </Route>
-    </Switch>
+    <BrowserRouter basename={window["getOpenmrsSpaBase"]()}>
+      <Switch>
+        <Route exact path="/patient/:patientUuid/chart/allergies">
```

Comment:
No we don't know that (URL). This is a URL related to patient-chart, but the extension may be used in other components (page), too. The same applies to all other extension components in this PR.

Ideally, the extension slot is supplied a patient UUID and a basePath. So you can derive your routes here via `${basePath}/allergies/...`.

Reply:
The `basePath` prop returned undefined so I just left it out. Once They're available, I'll refactor.

Reply:
Should be available.